canary/consumer-java/src/main/java/com/amazon/kinesis/video/canary/consumer/WebrtcStorageCanaryConsumer.java

```diff
@@ -0,0 +1,215 @@
+            datumList.add(datum);
+            MetricDatum aggDatum = new MetricDatum()
+                    .withMetricName(metricName)
+                    .withUnit(cwUnit)
+                    .withValue(value)
+                    .withDimensions(aggregatedDimension);
+            datumList.add(aggDatum);
+
+            PutMetricDataRequest request = new PutMetricDataRequest()
+                    .withNamespace("KinesisVideoSDKCanary")
+                    .withMetricData(datumList);
+            cwClient.putMetricData(request);
+        } catch (Exception e) {
+            log.error(e);
+        }
+    }
+
+    public static void main(final String[] args) throws Exception {
+
+        // Import configurable parameters.
+        final Integer canaryRunTime = Integer.parseInt(System.getenv("CANARY_DURATION_IN_SECONDS"));
+        streamName = System.getenv("CANARY_STREAM_NAME");
+        canaryLabel = System.getenv("CANARY_LABEL");
+        region = System.getenv("AWS_DEFAULT_REGION");
+
+        log.info("Stream name: {}", streamName);
+
+        canaryStartTime = new Date();
+
+        credentialsProvider = new EnvironmentVariableCredentialsProvider();
+        amazonKinesisVideo = AmazonKinesisVideoClientBuilder.standard()
+                .withRegion(region)
+                .withCredentials(credentialsProvider)
+                .build();
+        cwClient = AmazonCloudWatchClientBuilder.standard()
+                .withRegion(region)
+                .withCredentials(credentialsProvider)
+                .build();
+
+        Timer intervalMetricsTimer = new Timer("IntervalMetricsTimer");
+        TimerTask intervalMetricsTask;
+
+        switch (canaryLabel) {
+            case "WebrtcLongRunning": {
+                final CanaryFragmentList fragmentList = new CanaryFragmentList();
+                intervalMetricsTask = new TimerTask() {
+                    @Override
+                    public void run() {
+                        calculateFragmentContinuityMetric(fragmentList);
+                    }
+                };
+
+                // Initial delay of 30s to allow for ListFragment response to be populated
+                final long intervalInitialDelay = 30000;
+                final long intervalDelay = 20000; // NOTE: 16s interval was causing gaps in continuity, changing to 20s (2x the
+                                                  // typical fragment duration coming from media server)
+
+                // NOTE: Metric publishing will NOT begin if canaryRunTime is < intervalInitialDelay
+                intervalMetricsTimer.scheduleAtFixedRate(intervalMetricsTask, intervalInitialDelay, intervalDelay);
+                Thread.sleep(canaryRunTime * 1000);
+                intervalMetricsTimer.cancel();
+                break;
+            }
+            case "WebrtcPeriodic": {
+                while ((System.currentTimeMillis() - canaryStartTime.getTime()) < canaryRunTime * 1000) {
```

Comment:
Hmm don't see how it could be an infinite loop, can you please explain?

We do only need to calculate it once, but we need to keep retrying until there are frames returned by `GetMedia()`. `GetMedia()` will return after ~3 sec if no footage is available. Once footage is available, it will continue to visit frames as long as there is footage available. Adding a comment to explain this.